### ISSUE TITLE
Not able to apply multiple foreign key constraint steps

### ISSUE THREAD

**Issue by jacobg** [jOOQ/jOOQ#6379]
The following code does not compile:
```
				DSL.constraint("my_constraint")
						.foreignKey(TABLE1.FIELD)
						.references(TABLE2, TABLE2.FIELD)
						.onDeleteCascade()
						.onUpdateNoAction()
```
It looks like the ConstraintForeignKeyOnStep methods return ConstraintFinalStep type rather than its own ConstraintForeignKeyOnStep type to allow multiple configuration steps.

**Comment by lukaseder** [comment]
Thank you very much for your report. This is already fixed in jOOQ 3.10: https://github.com/jOOQ/jOOQ/issues/6129.

The workaround is this:

```java
				((ConstraintForeignKeyOnStep) DSL.constraint("my_constraint")
						.foreignKey(TABLE1.FIELD)
						.references(TABLE2, TABLE2.FIELD)
						.onDeleteCascade())
						.onUpdateNoAction()
```

I hope this helps

**Comment by jacobg** [comment]
Great thanks!
